library/src-bootstrapped/scala/internal/Quoted.scala

```diff
@@ -16,4 +16,7 @@ object Quoted {
   def typeQuote[T <: AnyKind]: Type[T] =
     throw new Error("Internal error: this method call should have been replaced by the compiler")
 
+  /** A splice in a quoted pattern is desugared by the compiler into a call to this method */
+  def patternHole[T]: T =
```

Comment:
Note: this is the equivalent to [`def hole[T]`](https://github.com/lampepfl/dotty/pull/6042/files#diff-c60b7bdb36081577552ac360877d74ebR22) in #6042, not to be confused with [`object patternHole { def unapply[T] = ... }`](https://github.com/lampepfl/dotty/pull/6042/files#diff-c60b7bdb36081577552ac360877d74ebR26).